autoconnect/autoconnect-ws/autoconnect-ws-sm/src/unidentified.rs

```diff
@@ -262,12 +264,12 @@ mod tests {
             db: hello_db().into_boxed_arc(),
             ..Default::default()
         });
-        let msg = ClientMessage::Hello {
-            uaid: None,
-            channel_ids: None,
-            use_webpush: Some(true),
-            broadcasts: None,
-        };
+        // Ensure that we do not need to pass the "use_webpush" flag.
+        // (yes, this could just be passing the string, but I want to be
+        // very explicit here.)
+        let json = serde_json::json!({"messageType":"hello"});
+        let raw = json.to_string();
+        let msg = ClientMessage::from_str(&raw).unwrap();
```

Comment:
This seems unnecessary with the type itself enforcing this -- it's very clear from the explicit construction of the struct variant in the previous code of what's actually being passed

Reply:
Right, but using the struct variant requires that fields be present, correct? Passing JSON like this would ensure that default values are being set by the constructor rather than being defined at initialization. This is essentially an even more stripped down version of the structure that's handed in with the above function where we specify all the fields as per the old format. Here, we don't include the previously required `use_webpush` as well as skip the `uaid` and rely on default management for the rest of the fields. 
This should be a properly accepted structure, correct? Shouldn't we test for that?